embeddings/transformation/hf_transformation/qa_data_split_transformation.py

```diff
@@ -0,0 +1,82 @@
+from typing import Optional
+
+import pandas as pd
+from datasets import Dataset, DatasetDict
+from sklearn.model_selection import train_test_split
+
+from embeddings.transformation.transformation import Transformation
+
+
+class QuestionAnsweringSplitsTransformation(Transformation[Dataset, DatasetDict]):
+    def __init__(
+        self,
+        train_size: float,
+        dev_size: float,
+        test_size: float,
+        seed: int,
+        context_column: str = "context",
+        stratify_column: Optional[str] = None,
```

Comment:
I would provide a default stratification column as context/document, otherwise, quite easily we can leak of the data - the same contexts will be in train / dev / test sets and it could make datasets easier to solve :d

Reply:
We can also write a docs on this issue.

Reply:
Maybe this parameter has wrong name, however it do not work as described. There won't be any leak in context, because the split it's based on contexts/documents, therefore it can't be in few splits at the same time. The stratification parameter was to handle equal distribution of documents of different datasets. Let's say we have `squad_v2` and `google_natural_question` merged dataset. There for we could stratify contexts based on the source dataset